tests/unit/api/multipartUpload.js

```diff
@@ -1626,3 +1667,114 @@ describe('Multipart Upload API', () => {
         });
     });
 });
+
+describe('complete mpu with versioning', () => {
+    const objData = ['foo0', 'foo1', 'foo2'].map(str =>
+        Buffer.from(str, 'utf8'));
+
+    function _createBucketPutVersioningReq(status) {
+        const request = {
+            bucketName,
+            headers: {
+                host: `${bucketName}.s3.amazonaws.com`,
+            },
+            url: '/?versioning',
+            query: { versioning: '' },
+        };
+        const xml = '<VersioningConfiguration ' +
+        'xmlns="http://s3.amazonaws.com/doc/2006-03-01/">' +
+        `<Status>${status}</Status>` +
+        '</VersioningConfiguration>';
+        request.post = xml;
+        return request;
+    }
+
+    function _createPutObjectRequest(body) {
+        const params = {
+            bucketName,
+            namespace,
+            objectKey,
+            headers: {},
+            url: `/${bucketName}/${objectKey}`,
+        };
+        return new DummyRequest(params, body);
+    }
+
+    const enableVersioningRequest = _createBucketPutVersioningReq('Enabled');
+    const suspendVersioningRequest = _createBucketPutVersioningReq('Suspended');
+    const testPutObjectRequests = objData.slice(0, 2).map(data =>
+        _createPutObjectRequest(data));
+
+    function _assertDataStoreValues(expectedValues) {
+        assert.strictEqual(ds.length, expectedValues.length + 1);
+        for (let i = 0, j = 1; i < expectedValues.length; i++, j++) {
+            if (expectedValues[i] === undefined) {
+                assert.strictEqual(ds[j], expectedValues[i]);
+            } else {
+                assert.deepStrictEqual(ds[j].value, expectedValues[i]);
+            }
+        }
+    }
+
+    before(done => {
+        cleanup();
+        async.series([
```

Comment:
I like to use series when I don't need the return arguments for the following call (e.g. I can just have `callback` as the parameter for each function, unlike a waterfall). In this case I'm comparing the expected result against `ds` rather than any of the results of the functions, so I don't need the array of results object.